src/lib/common/network_config_logger.cpp

```diff
@@ -0,0 +1,44 @@
+#include "network_config_logger.h"
+#include "FreeRTOS.h"
+#include "bridgeLog.h"
+#include "cbor.h"
+
+static char *log_buffer = NULL;
+static size_t log_buffer_size = 0;
+static constexpr size_t log_buffer_max_size = 2048;
+
+static CborError print_stream_handler(void *out, const char *fmt, ...) {
+  configASSERT(log_buffer != NULL);
+  (void)out;
+
+  va_list args;
+  va_start(args, fmt);
+  int n = vsnprintf(nullptr, log_buffer_max_size - log_buffer_size, fmt, args);
+  if (n > 0) {
+    vsnprintf(&log_buffer[log_buffer_size], log_buffer_max_size - log_buffer_size, fmt, args);
```

Comment:
We no longer call `vBridgeLogPrint` for every call to this function, which occurs almost for every character. Instead we build the log line in a heap-allocated buffer.